examples/c/profile.c

```diff
@@ -149,13 +150,21 @@ int main(int argc, char *const argv[])
 	int argp, i, err = 0;
 	bool *online_mask = NULL;
 
-	while ((argp = getopt(argc, argv, "hf:")) != -1) {
+	static struct option long_options[] = {
+		{"sw-event", no_argument, 0, 1},
+		{0, 0, 0, 0}
+	};
+
+	while ((argp = getopt_long(argc, argv, "hf:", long_options, NULL)) != -1) {
 		switch (argp) {
 		case 'f':
 			freq = atoi(optarg);
 			if (freq < 1)
 				freq = 1;
 			break;
+		case 1:
```

Comment:
Nit: This `1` seems pretty magic-y. I had to page in `getopt_long` behavior. Perhaps we could use `'s'` or something instead of `1`, and/or introduce a constant of sorts?

Reply:
I changed `1` to `s`.